domain/bakerystorage/state/config.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/canonical/sqlair"
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery"
+	"github.com/juju/errors"
+
+	coredatabase "github.com/juju/juju/core/database"
+	"github.com/juju/juju/domain"
+	internaldatabase "github.com/juju/juju/internal/database"
+)
+
+type ConfigState struct {
+	*domain.StateBase
+}
+
+func NewConfigState(factory coredatabase.TxnRunnerFactory) *ConfigState {
+	return &ConfigState{
+		StateBase: domain.NewStateBase(factory),
+	}
+}
+
+func (st *ConfigState) InitialiseBakeryConfig(
+	ctx context.Context,
+	localUsersKey,
+	localUsersThirdPartyKey,
+	externalUsersThirdPartyKey,
+	offersThirdPartyKey *bakery.KeyPair,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	cfg := Config{
+		LocalUsersPrivateKey:              &keyScanner{localUsersKey.Private.Key},
+		LocalUsersPublicKey:               &keyScanner{localUsersKey.Public.Key},
+		LocalUsersThirdPartyPrivateKey:    &keyScanner{localUsersThirdPartyKey.Private.Key},
+		LocalUsersThirdPartyPublicKey:     &keyScanner{localUsersThirdPartyKey.Public.Key},
+		ExternalUsersThirdPartyPrivateKey: &keyScanner{externalUsersThirdPartyKey.Private.Key},
+		ExternalUsersThirdPartyPublicKey:  &keyScanner{externalUsersThirdPartyKey.Public.Key},
+		OffersThirdPartyPrivateKey:        &keyScanner{offersThirdPartyKey.Private.Key},
+		OffersThirdPartyPublicKey:         &keyScanner{offersThirdPartyKey.Public.Key},
+	}
+
+	initialiseConfigStmt, err := sqlair.Prepare("INSERT INTO bakery_config (*) VALUES ($Config.*)", Config{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, initialiseConfigStmt, cfg).Run()
+		if internaldatabase.IsErrConstraintUnique(err) {
+			return BakeryConfigAlreadyInitialised
+		}
+		return domain.CoerceError(err)
+	})
+	return errors.Trace(err)
+}
+
+func (st *ConfigState) GetLocalUsersKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return &bakery.KeyPair{
+		Public:  bakery.PublicKey{Key: cfg.LocalUsersPublicKey.Key()},
+		Private: bakery.PrivateKey{Key: cfg.LocalUsersPrivateKey.Key()},
+	}, nil
+}
+
+func (st *ConfigState) GetLocalUsersThirdPartyKey(ctx context.Context) (*bakery.KeyPair, error) {
+	cfg, err := st.getConfig(ctx)
```

Comment:
Have each key pair extracted per method. We don't need a getConfig, it forces us to do more work than we require. Repetition is better in this case.

Reply:
Done